### PR TITLE
Show V2 Pay Button wallet connect and switch network

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Shows the correct text and allows users to connect wallet on pay.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1798** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1798" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1798?utm_source=stack-comment).

File: src/components/TransactionModal.tsx [hunk 1]
```diff
@@ -53,20 +55,32 @@ const PendingTransactionModalBody = () => {
...
-  const okText = userAddress
-    ? props.okText
-    : props.connectWalletText ?? t`Connect wallet`
+  const { isConnected, chainUnsupported } = useWallet()
+  const okText = useMemo(() => {
+    if (chainUnsupported) {
+      return props.switchNetworkText ?? `Switch network to ${readNetwork.label}`
+    }
+    if (!isConnected) {
+      return props.connectWalletText ?? 'Connect wallet'
```
Comment (johnnyd-eth) [inline src/components/TransactionModal.tsx:64]: **Issue:** ```t`Connect wallet` ```

File: src/components/TransactionModal.tsx [hunk 2]
```diff
@@ -53,20 +55,32 @@ const PendingTransactionModalBody = () => {
...
  */
 export default function TransactionModal(props: TransactionModalProps) {
-  const { userAddress } = useWallet()
-  const okText = userAddress
-    ? props.okText
-    : props.connectWalletText ?? t`Connect wallet`
+  const { isConnected, chainUnsupported } = useWallet()
+  const okText = useMemo(() => {
+    if (chainUnsupported) {
+      return props.switchNetworkText ?? `Switch network to ${readNetwork.label}`
```
Comment (johnnyd-eth) [inline src/components/TransactionModal.tsx:61]: ```t``Switch network to ${readNetwork.label}````

**Comment by johnnyd-eth** [comment]
Very VERY dope. Just one comment then LFG

**Review by johnnyd-eth — CHANGES_REQUESTED** [review]
.

File: src/components/TransactionModal.tsx (hunk 2 above)
Comment (wraeth-eth) [reply src/components/TransactionModal.tsx:61]: good catch 🙏

**Review by johnnyd-eth — APPROVED** [review]
LGTM

**Pull request merged**